identity_stardust/src/stardust_document.rs

```diff
@@ -51,8 +50,46 @@ impl StardustDocument {
     )
   }
 
+  /// Temporary testing implementation.
+  pub fn tmp_add_verification_method(
+    &mut self,
+    id: CoreDID,
+    keypair: &KeyPair,
+    fragment: &str,
+    scope: MethodScope,
+  ) -> Result<()> {
+    let method: VerificationMethod = VerificationMethod::new(id, keypair.type_(), keypair.public(), fragment)?;
+
+    self.0.insert_method(method, scope)?;
+
+    Ok(())
+  }
+
+  /// Temporary testing implementation.
+  pub fn tmp_add_service(&mut self, id: CoreDID, fragment: &str, type_: &str, endpoint: ServiceEndpoint) -> Result<()> {
+    let service = Service::builder(Object::new())
+      .type_(type_)
+      .id(id.join(fragment).unwrap())
+      .service_endpoint(endpoint)
+      .build()?;
+
+    self.0.service_mut().append(service);
+
+    Ok(())
+  }
+
+  /// Temporary testing implementation.
+  pub fn tmp_set_id(&mut self, mut id: CoreDID) {
+    std::mem::swap(self.0.id_mut(), &mut id);
+  }
+
+  /// Temporary testing implementation.
+  pub fn tmp_id(&self) -> &CoreDID {
+    self.0.id()
+  }
+
   /// Returns the placeholder DID of newly constructed DID Documents,
-  /// `"did:stardust:0000000000000000000000000000000000000000000000000000000000000000"`.
+  /// `"did:0:0"`.
   // TODO: generalise to take network name?
   pub fn placeholder_did() -> &'static CoreDID {
     &PLACEHOLDER_DID
```

Comment:
This only works because `StardustDocument` is using `CoreDID`. When `StardustDID` is a thing, `did:0:0` will not pass validation checks on creation (which is fine, since it's only replaced during serialization).

Reply:
For this case, we can introduce another `STARDUST_PLACHOLDER_DID` or something that is the minimally valid stardust DID. Since the replacement procedure doesn't care about the did of self, that should work. I'd leave it for now and leave this to the implementation of `StardustDID`.